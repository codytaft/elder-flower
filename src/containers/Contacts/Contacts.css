.sos-btn {
  margin: 2rem auto 2rem auto;
  text-decoration: none;
  justify-content: center;
  padding: 2rem;
  border: 0.2rem solid #4e2d1d;
  color: #453d30;
  border-radius: 5rem;
  cursor: pointer;
  width: 30rem;
  background: #eacda3;
  font-size: 32px;
  font-family: 'Nunito', sans-serif; }

.sos-btn:focus {
  outline: none; }

.sos-btn:hover {
  background-color: #c3fcf2; }

.sos-btn-hidden {
  display: none; }
