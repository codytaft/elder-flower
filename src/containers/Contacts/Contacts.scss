.sos-btn {
  margin: 2rem auto 2rem auto;
  text-decoration: none;
  justify-content: center;
  padding: 2rem;
  border: 0.2rem solid rgb(78, 45, 29);
  color: #453d30;
  border-radius: 5rem;
  cursor: pointer;
  width: 30rem;
  background: #eacda3;
  font-size: 32px;
  font-family: 'Nunito', sans-serif;
}

.sos-btn:focus {
  outline: none;
}

.sos-btn:hover {
  background-color: #c3fcf2;
}

.sos-btn-hidden {
  display: none;
}

.question-input {
  margin: 1rem;

  // margin: 2rem auto 2rem auto;
  text-decoration: none;
  justify-content: center;
  padding: 1rem 1.5rem;
  border: 0.2rem solid rgb(78, 45, 29);
  color: #453d30;
  border-radius: 5rem;
  width: 15rem;
  font-size: 32px;
  font-family: 'Nunito', sans-serif;
  height: auto;
}

.carer-question-submit {
  text-align: left;
  justify-content: center;
  margin: 1rem;
  padding: 1rem 1.5rem;
  border: 0.2rem solid rgb(78, 45, 29);
  color: #453d30;
  border-radius: 5rem;
  cursor: pointer;
  width: auto;
  background: #eacda3;
  font-size: 32px;
  font-family: 'Nunito', sans-serif;
  height: auto;
}

.carer-question-submit:hover {
  background: #c3fcf2;
}

.question-form {
  display: flex;
  justify-content: center;
  align-content: center;
  align-items: center;
}
