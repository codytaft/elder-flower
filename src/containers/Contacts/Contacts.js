import React, { Component } from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { sendSOS, textQuestion } from '../../helpers/fetchCalls';
import './Contacts.css';

export class Contacts extends Component {
  constructor() {
    super();
    this.state = {
      question: ''
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { currentUser } = this.props;
    textQuestion(
      currentUser.contactPhone,
      currentUser.contactName,
      this.state.question
    );
  };
  render() {
    const { currentUser } = this.props;
    return (
      <div>
        {currentUser ? (
          <div>
            {currentUser.isElder ? (
              <div>
                <h1>Your Carer Name: {currentUser.contactName}</h1>
                <h1>Your Carer Phone Number: {currentUser.contactPhone}</h1>
                <button
                  onClick={() =>
                    sendSOS(currentUser.contactPhone, currentUser.contactName)
                  }
                  className="sos-btn"
                >
                  Send SOS Message
                </button>
              </div>
            ) : (
              <div>
                <h1>Your Elder Name: {currentUser.contactName}</h1>
                <h1>Your Elder Phone Number: {currentUser.contactPhone}</h1>
                <form onSubmit={this.handleSubmit}>
                  <input
                    onChange={this.handleChange}
                    type="text"
                    name="question"
                    value={this.state.question}
                    className="question-input"
                    placeholder="Ask question"
                  />
                  <button className="carer-question-submit">
                    Text Question
                  </button>
                </form>
              </div>
            )}
          </div>
        ) : (
          <div>
            <h1>Please Login</h1>
            <NavLink exact to="/login" className="top-nav-link nav-link">
              Login
            </NavLink>
          </div>
        )}
      </div>
    );
  }
}

Contacts.propTypes = {
  location: PropTypes.object,
  currentUser: PropTypes.object
};

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export default withRouter(
  connect(
    mapStateToProps,
    null
  )(Contacts)
);
