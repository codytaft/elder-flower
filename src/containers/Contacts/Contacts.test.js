import React from 'react';
import { shallow } from 'enzyme';
import { Contacts, mapStateToProps } from './Contacts';

describe('Contacts', () => {
  it.skip('should match the snapshot', () => {
    let wrapper;
    let mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };

    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockCurrentUser)
      })
    );

    wrapper = shallow(<Contacts currentUser={mockCurrentUser} />);
    wrapper.find('button').simulate('click');
    expect(wrapper).toMatchSnapshot();
    expect(window.fetch).toHaveBeenCalledWith(
      mockCurrentUser.contactName,
      mockCurrentUser.contactPhone
    );
  });
});

describe('mapStateToProps', () => {
  it('should return an object with a currentUser object', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockState = {
      currentUser: { ...mockCurrentUser }
    };

    const expected = {
      currentUser: {
        id: 75,
        firstName: 'Cody',
        lastName: 'Taft',
        phoneNumber: '+19038511575',
        email: 'cody.taft@gmail.com',
        password: '123',
        contactName: 'Gaynell',
        contactPhone: '+17203304593',
        isElder: false
      }
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
