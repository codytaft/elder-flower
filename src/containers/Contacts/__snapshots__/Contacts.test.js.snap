// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contacts
    currentUser={
        Object {
            "contactName": "Gaynell",
            "contactPhone": "+17203304593",
            "email": "cody.taft@gmail.com",
            "firstName": "Cody",
            "id": 75,
            "isElder": false,
            "lastName": "Taft",
            "password": "123",
            "phoneNumber": "+19038511575",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div>
                <h1>
                        Your Elder Name: 
                        Gaynell
                </h1>
                <h1>
                        Your Elder Phone Number: 
                        +17203304593
                </h1>
                <form
                        onSubmit={[Function]}
                >
                        <input
                                className="question-input"
                                name="question"
                                onChange={[Function]}
                                placeholder="Ask question"
                                type="text"
                                value=""
                        />
                        <button
                                className="carer-question-submit"
                        >
                                Text Question
                        </button>
                </form>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
                    Your Elder Name: 
                    Gaynell
          </h1>
          <h1>
                    Your Elder Phone Number: 
                    +17203304593
          </h1>
          <form
                    onSubmit={[Function]}
          >
                    <input
                              className="question-input"
                              name="question"
                              onChange={[Function]}
                              placeholder="Ask question"
                              type="text"
                              value=""
                    />
                    <button
                              className="carer-question-submit"
                    >
                              Text Question
                    </button>
          </form>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Your Elder Name: 
              Gaynell
</h1>,
            <h1>
              Your Elder Phone Number: 
              +17203304593
</h1>,
            <form
              onSubmit={[Function]}
>
              <input
                            className="question-input"
                            name="question"
                            onChange={[Function]}
                            placeholder="Ask question"
                            type="text"
                            value=""
              />
              <button
                            className="carer-question-submit"
              >
                            Text Question
              </button>
</form>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Your Elder Name: ",
                "Gaynell",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Your Elder Name: ",
              "Gaynell",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Your Elder Phone Number: ",
                "+17203304593",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Your Elder Phone Number: ",
              "+17203304593",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="question-input"
                  name="question"
                  onChange={[Function]}
                  placeholder="Ask question"
                  type="text"
                  value=""
/>,
                <button
                  className="carer-question-submit"
>
                  Text Question
</button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "question-input",
                  "name": "question",
                  "onChange": [Function],
                  "placeholder": "Ask question",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Text Question",
                  "className": "carer-question-submit",
                },
                "ref": null,
                "rendered": "Text Question",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div>
                    <h1>
                              Your Elder Name: 
                              Gaynell
                    </h1>
                    <h1>
                              Your Elder Phone Number: 
                              +17203304593
                    </h1>
                    <form
                              onSubmit={[Function]}
                    >
                              <input
                                        className="question-input"
                                        name="question"
                                        onChange={[Function]}
                                        placeholder="Ask question"
                                        type="text"
                                        value=""
                              />
                              <button
                                        className="carer-question-submit"
                              >
                                        Text Question
                              </button>
                    </form>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <h1>
                        Your Elder Name: 
                        Gaynell
            </h1>
            <h1>
                        Your Elder Phone Number: 
                        +17203304593
            </h1>
            <form
                        onSubmit={[Function]}
            >
                        <input
                                    className="question-input"
                                    name="question"
                                    onChange={[Function]}
                                    placeholder="Ask question"
                                    type="text"
                                    value=""
                        />
                        <button
                                    className="carer-question-submit"
                        >
                                    Text Question
                        </button>
            </form>
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Your Elder Name: 
                Gaynell
</h1>,
              <h1>
                Your Elder Phone Number: 
                +17203304593
</h1>,
              <form
                onSubmit={[Function]}
>
                <input
                                className="question-input"
                                name="question"
                                onChange={[Function]}
                                placeholder="Ask question"
                                type="text"
                                value=""
                />
                <button
                                className="carer-question-submit"
                >
                                Text Question
                </button>
</form>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Your Elder Name: ",
                  "Gaynell",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Your Elder Name: ",
                "Gaynell",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Your Elder Phone Number: ",
                  "+17203304593",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Your Elder Phone Number: ",
                "+17203304593",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="question-input"
                    name="question"
                    onChange={[Function]}
                    placeholder="Ask question"
                    type="text"
                    value=""
/>,
                  <button
                    className="carer-question-submit"
>
                    Text Question
</button>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "question-input",
                    "name": "question",
                    "onChange": [Function],
                    "placeholder": "Ask question",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Text Question",
                    "className": "carer-question-submit",
                  },
                  "ref": null,
                  "rendered": "Text Question",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
